code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/LegacyNotificationBuilderTest.kt

```diff
@@ -0,0 +1,161 @@
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.builders
+
+import android.app.Activity
+import android.content.Context
+import android.os.Build
+import android.os.Bundle
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants.DefaultValues.DEFAULT_CHANNEL_ID
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants.DefaultValues.SILENT_NOTIFICATION_CHANNEL_ID
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.BasicPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MockAEPPushTemplateDataProvider
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.removeKeysFromMap
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.replaceValueInMap
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.IntentData
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.MapData
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertNotNull
+import junit.framework.TestCase.assertNull
+import org.junit.Assert.assertArrayEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.Shadows
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class LegacyNotificationBuilderTest {
+
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var context: Context
+    private lateinit var dataMap: MutableMap<String, String>
+    private lateinit var mockBundle: Bundle
+
+    @Before
+    fun setUp() {
+        context = RuntimeEnvironment.getApplication()
+        trackerActivityClass = DummyActivity::class.java
+        dataMap = MockAEPPushTemplateDataProvider.getMockedAEPDataMapWithAllKeys()
+        mockBundle = MockAEPPushTemplateDataProvider.getMockedAEPBundleWithAllKeys()
+    }
+
+    @Test
+    fun `verify construct should map valid BasicPushTemplate data to notification data`() {
+        val pushTemplate = BasicPushTemplate(MapData(dataMap))
+        val notificationBuilder =
+            LegacyNotificationBuilder.construct(context, pushTemplate, trackerActivityClass)
+        val pendingIntent = notificationBuilder.build().contentIntent
+        val shadowPendingIntent = Shadows.shadowOf(pendingIntent)
+        val intent = shadowPendingIntent.savedIntent
+
+        assertEquals(NotificationCompat.Builder::class.java, notificationBuilder.javaClass)
+        assertEquals(pushTemplate.ticker, notificationBuilder.build().tickerText)
+        assertEquals(
+            pushTemplate.title,
+            notificationBuilder.build().extras.getString(NotificationCompat.EXTRA_TITLE)
```

Comment:
Can you store the built notification in a variable instead of calling `notificationBuilder.build()` every time it is needed. Or, is there a limitation that needs this approach?

Reply:
done